wg-naming/README.md

```diff
@@ -0,0 +1,70 @@
+<!---
+This is an autogenerated file!
+
+Please do not edit this file directly, but instead make changes to the
+sigs.yaml file in the project root.
+
+To understand how this file is generated, see https://git.k8s.io/community/generator/README.md
+--->
+# Naming Working Group
+
+
+## Stakeholder SIGs
+* SIG Architecture
+* SIG Contributor Experience
+* SIG Docs
+
+
+
+## Organizers
+
+* Celeste Horgan (**[@celestehorgan](https://github.com/celestehorgan)**), CNCF
+* Stephen Augustus (**[@justaugustus](https://github.com/justaugustus)**), VMware
+* Zach Corleissen (**[@zacharysarah](https://github.com/zacharysarah)**), Linux Foundation
+
+## Contact
+- Slack: [#wg-naming](https://kubernetes.slack.com/messages/wg-naming)
+- [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-wg-naming)
+- [Open Community Issues/PRs](https://github.com/kubernetes/community/labels/wg%2Fnaming)
+<!-- BEGIN CUSTOM CONTENT -->
+
+**The following section will be reworked and formalized as a charter once the
+Working Group has been approved by the Steering Committee.**
+
+### Goals
+
+- Evaluate language and naming choices within the Kubernetes organization, with
+  a specific initial focus of:
+  - Removing language which cannot be fully separated from racist, sexist or
+    homophobic connotations in cultural contexts outside of computer
+    programming
+  - Improving clarity of codebases and documentation by replacing idioms,
+    metaphors and slang specific to the English language
+- Create a list of terms to avoid and propose alternatives
+- Define a process by which any member of the Kubernetes organization can
+  propose a language recommendation, have them evaluated based on a rubric, and
+  see them implemented across all codebases.
+  - Provide an easily findable location for language recommendations and
+    follow-up issues, similar to an architectural decision record.
+- Work with stakeholder SIGs to implement the changes recommended. We
+  anticipate the following:
+  - Provide stakeholder SIGs with guidance on naming, language conventions, and
+    processes
+  - Collaborate with SIG Architecture and other stakeholders on an
+    implementation timeline and strategies for dealing with follow-up issues
+    from renaming, like deprecations
+  - Collaborate with SIG Docs and SIG Contributor Experience on documenting
+    language recommendations and processes
+  - Work with the Code of Conduct Committee to add code architecture to the COC
```

Comment:
I'm not very clear on what this point means.

Reply:
Neither am I. @justaugustus @celestehorgan @zacharysarah Could you elaborate on this point?

Reply:
@deads2k @cblecker The goal is for inclusive naming in software architecture, not just in docs. For example, replacing "master/slave" terminology in docs but not in functions, classes, variables, etc. is an empty gesture. 

What's a more accurate way to refer to code in this specific context?

Reply:
We may need the full @kubernetes/code-of-conduct-committee to weigh in on this. One could argue that code is language as much as an issue or Slack conversation. That would cover future cases, but remediation of existing code would have to be handled by individuals in the OWNERS files IMO, much like if someone trashes a GitHub issue, we (the CoC Committee) ask the Kubernetes GitHub admin team to remove the offending comments. I still believe the existing CoC covers this, as we collectively define the edges of what is acceptable or not. From the CoC specifically:

> We are committed to making participation in this project a harassment-free experience for everyone, regardless of level of experience, gender, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, or nationality.

Reply:
Yeah, I don't think the CoC needs to be modified here. Perhaps this goal can be changed to "Review with CoCC guidelines and polices around use of language/terminology in code" or something similar?

The idea being:
- I honestly don't believe any of the terminology currently used in code was placed there intentionally to harass anyone (if this was the case, it would already be CoC covered)
- We still want to be able to make binding policies around how to remediate language used in code, to ensure it doesn't bring harm. How do we do that? (I agree with the binding part, but I'm not sure the CoC is the right tool)

But bringing it back to how to we approve this WG, I'd suggest changing this to "review" as I've suggested above will allow the freedom to explore this question throughly with the CoCC to answer if the CoC is or is not the right tool to address this.

Reply:
For now, I suggest we remove any mention of the COCC from this WG, and put it as an agenda item for the WG's work. 

That said, per later discussion in this thread: @jdumars is joining us in this initiative, so involvement from that quarter will be there.

Reply:
(Added as a post-formation discussion point.)